src/com/toopher/openam/ToopherSecondFactor.java

```diff
@@ -73,154 +68,80 @@ private void debug_message(String message) {
 
     @Override
     public void init(Subject subject, Map sharedState, Map options) {
-        super.init(subject, sharedState, options);
+        if (isUseFirstPassEnabled()) {
+            this.sharedState = sharedState;
+        }
+        if (sharedState == null) {
+            debug_message("sharedState is null!");
+        }
+        if (options == null) {
+            debug_message("options is null");
+        }
+        this.options = options;
```

Comment:
actually, these debug messages are just leftover cruft from when I didn't really know what I was doing in OpenAM.  I'll nuke them.